test/runtests.jl

```diff
@@ -817,3 +818,72 @@ end
     @test r.i isa UInt16
     @test r.i == 1
 end
+
+#####
+##### constraints
+#####
+
+@schema "test.constraint" Constraint
+
+const CONSTRAINT_V1_EQUAL_CONSTRAINT_LINE = @__LINE__() + 4
+@version ConstraintV1 begin
+    a
+    b = clamp(b, 0, 5)
+    @check a == b
+    @check a > 0
+end
+
+@testset "constraints" begin
+    r = ConstraintV1(; a=1, b=1)
+    @test r isa ConstraintV1
+    @test r.a === 1
+    @test r.b === 1
+
+    r = ConstraintV1(; a=1, b=1.0)
+    @test r isa ConstraintV1
+    @test r.a === 1
+    @test r.b === 1.0
+
+    # In Julia 1.8+ we can test can test against "CheckConstraintError: a == b"
+    try
+        ConstraintV1(; a=1, b=2)
+        @test false
+    catch e
+        @test e isa CheckConstraintError
+        @test e.predicate == :(a == b)
+    end
+
+    try
+        ConstraintV1(; a=0, b=0)
+        @test false
+    catch e
+        @test e isa CheckConstraintError
+        @test e.predicate == :(a > 0)
+    end
+
+    try
+        ConstraintV1(; a=6, b=6)
+        @test false
+    catch e
+        @test e isa CheckConstraintError
+        @test e.predicate == :(a == b)
+    end
+
+    # For exceptions that occur during processing constraints its convenient to include the
+    # location of the `@check` in the stacktrace.
+    try
+        ConstraintV1(; a=1, b=missing)  # Fails when running check `a == b`
+        @test false
```

Comment:
??

Reply:
Ensures that if `ConstraintV1(; a=1, b=missing)` does not thrown an exception that a test failure is reported.

Reply:
ah okay, could use a comment too.